### PR TITLE
Add citation/copyright file, plus MIT licence

### LINKED ISSUE #3: Licence
stella needs a licence, I recommend either [MIT](https://opensource.org/licenses/MIT) or [BSD-3-Clause](https://opensource.org/licenses/BSD-3-Clause). Both essentially say "do whatever you want, but you must credit the authors". GS2 is MIT-licensed, so that's probably the best starting point. The MIT licence does allow changing the licence, so we're not limited to that.

### PR DESCRIPTION
**Pull request by ZedThree**
Resolves #3 

@mabarnes @DenSto I've been a bit presumptive and assumed you're alright with the MIT licence.

CITATION.cff is in the [Citation File Format](https://citation-file-format.github.io), and basically is a way of letting people cite the _software_. It can also reference e.g. the stella paper.

I've included GS2 authors from before 2018 with `git log --reverse --pretty=tformat:'%an' --date=short --before 2018 | sort | uniq`. @mabarnes Let me know if that should be different.

### CODE REVIEW

**Comment by mabarnes** [comment]
Am fine with MIT license.  The author list is not accurate, as any ported code will have come from a much older version of gs2/agk (closer to 2011 than 2018).  Not sure how important it is to have this right, but I can at least remove those from the list who have definitely not contributed to any of the stella code/algorithm and add anyone who might have contributed since.

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  - family-names: Jackson
+    given-names: Adrian
+
+  - family-names: Kotschenreuther
+    given-names: Michael T.
+    affiliation: Institute for Fusion Studies, University of Texas
+
+  - family-names: Lee
+    given-names: Jungpyo
+
```
Comment (mabarnes) [inline CITATION.cff:97]: suggest removing

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  - family-names: Anton
+    given-names: Lucian
+
+  - family-names: Ball
+    given-names: Justin
+
+  - family-names: Baumgaertel
+    given-names: Jessica
+
```
Comment (mabarnes) [inline CITATION.cff:71]: suggest removing

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  - family-names: Parisi
+    given-names: Jason
+    affiliation: University of Oxford
+    orcid: https://orcid.org/0000-0003-1328-7154
+
+  - family-names: Wilkie
+    given-names: George
+    orcid: https://orcid.org/0000-0002-6826-6273
+
```
Comment (mabarnes) [inline CITATION.cff:62]: suggest removing

File: CITATION.cff [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  - family-names: Lee
+    given-names: Jungpyo
+
+  - family-names: Leggate
+    given-names: Huw
+
+  - family-names: Mandell
+    given-names: Noah
+
```
Comment (mabarnes) [inline CITATION.cff:103]: suggest removing

File: CITATION.cff [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  - family-names: Mauriya
+    given-names: Adwiteey
+
```
Comment (mabarnes) [inline CITATION.cff:106]: suggest removing

File: CITATION.cff [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  - family-names: Tatsuno
+    given-names: Tomo
+
+  - family-names: Van Wyk
+    given-names: Ferdinand
+
```
Comment (mabarnes) [inline CITATION.cff:112]: suggest removing

File: CITATION.cff [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  - family-names: Wilkie
+    given-names: George
+    orcid: https://orcid.org/0000-0002-6826-6273
+
+  - family-names: Anton
+    given-names: Lucian
+
+  - family-names: Ball
+    given-names: Justin
+
```
Comment (mabarnes) [inline CITATION.cff:68]: suggest removing

File: CITATION.cff [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+  - family-names: Baumgaertel
+    given-names: Jessica
+
+  - family-names: Colyer
+    given-names: Greg
+
```
Comment (mabarnes) [inline CITATION.cff:74]: suggest removing

File: CITATION.cff [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+  - family-names: Hardman
+    given-names: Michael
+
```
Comment (mabarnes) [inline CITATION.cff:77]: suggest removing

File: CITATION.cff [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+  - family-names: Hein
+    given-names: Joachim
+
+  - family-names: Highcock
+    given-names: Edmund
+    orcid: https://orcid.org/0000-0002-3398-817X
+
+  - family-names: Howes
+    given-names: Gregory
+
```
Comment (mabarnes) [inline CITATION.cff:87]: suggest removing

File: CITATION.cff [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+  - family-names: Kotschenreuther
+    given-names: Michael T.
+    affiliation: Institute for Fusion Studies, University of Texas
+
+  - family-names: Lee
+    given-names: Jungpyo
+
+  - family-names: Leggate
+    given-names: Huw
+
```
Comment (mabarnes) [inline CITATION.cff:100]: suggest removing

File: CITATION.cff [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+  - family-names: Barnes
+    given-names: Michael
+    affiliation: University of Oxford
+    orcid: https://orcid.org/0000-0002-0177-1689
+
+  - family-names: St-Onge
+    given-names: Denis
+    affiliation: University of Oxford
+    orcid: https://orcid.org/0000-0003-3112-9221
+
```
Comment (mabarnes) [inline CITATION.cff:19]: suggest adding Alexander von Boetticher, affiliated with University of Oxford and Robert Davies at York, as well as Jose Manuel Garcia-Regana at CIEMAT.  There may be others as well, but we can sort those out as they commit

File: CITATION.cff [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+  - family-names: Colyer
+    given-names: Greg
+
+  - family-names: Hardman
+    given-names: Michael
+
+  - family-names: Hein
+    given-names: Joachim
+
```
Comment (mabarnes) [inline CITATION.cff:80]: suggest removing

File: CITATION.cff [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+    affiliation: University of Oxford
+    orcid: https://orcid.org/0000-0003-1328-7154
+
+  - family-names: Wilkie
+    given-names: George
+    orcid: https://orcid.org/0000-0002-6826-6273
+
+  - family-names: Anton
+    given-names: Lucian
+
```
Comment (mabarnes) [inline CITATION.cff:65]: suggest removing

File: CITATION.cff [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+  - family-names: Highcock
+    given-names: Edmund
+    orcid: https://orcid.org/0000-0002-3398-817X
+
+  - family-names: Howes
+    given-names: Gregory
+
+  - family-names: Jackson
+    given-names: Adrian
+
```
Comment (mabarnes) [inline CITATION.cff:90]: suggest removing

**Review by mabarnes — CHANGES_REQUESTED** [review]
I've suggested removing authors who I'm 99% sure did not contribute to either the algorithms or the actual code in stella and suggested adding a few authors who have definitely contributed or are in the process of doing so.

File: CITATION.cff (hunk 9 above)
Comment (ZedThree) [reply CITATION.cff:77]: He does have one commit in stella: c9e3eb4

Comment (DenSto) [reply CITATION.cff:77]: This seems like just some changes to the makefile to fix compilation, and I may have told him to make these changes.

File: CITATION.cff (hunk 12 above)
Comment (ZedThree) [reply CITATION.cff:19]: Done

**Comment by ZedThree** [comment]
I've rebased and force-pushed this, which I wouldn't normally recommend, in order to make it unambiguous that the initial list of authors was incorrect.

**Pull request merged**